sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py

```diff
@@ -367,15 +330,34 @@ def __str__(self):
                 ('u' if self.update else '') +
                 ('p' if self.process else ''))
 
-
-AccountPermissions.READ = AccountPermissions(read=True)
-AccountPermissions.WRITE = AccountPermissions(write=True)
-AccountPermissions.DELETE = AccountPermissions(delete=True)
-AccountPermissions.LIST = AccountPermissions(list=True)
-AccountPermissions.ADD = AccountPermissions(add=True)
-AccountPermissions.CREATE = AccountPermissions(create=True)
-AccountPermissions.UPDATE = AccountPermissions(update=True)
-AccountPermissions.PROCESS = AccountPermissions(process=True)
+    @classmethod
+    def from_string(cls, permission):
+        if len(permission) > 8:
+            raise ValueError("Invalid Permission String")
+        read, write, delete, list = False, False, False, False
+        add, create, update, process = False, False, False, False
+        curr = 0
+        while curr < len(permission):
```

Comment:
You don't need to do this because strings are iterable in Python:
```
for letter in permission_str:
    read = (letter == 'r')
```
or you could just use the same logic that's in the constructor:
```
read = ('r' in permission)
write = ('w' in permission)
...
```